src/main/java/org/mockito/internal/util/collections/ListUtil.java

```diff
@@ -12,8 +12,10 @@
  * use cases neatly. However, we want to keep Mockito dependencies minimal. In Java8 we should be
  * able to get rid of this class.
  */
+@Deprecated
 public final class ListUtil {
```

Comment:
For my understanding, are there any remaining usages of `ListUtil`? If not, then we can remove them. This is an internal class and we don't allow users on relying on it. The only reason I wanted to keep all of this was to remain compatible with our users who have to run on older versions of Android, but it seems like our CI is happy with this PR. Therefore, I think we can remove this altogether, if we removed all of our own usages.

Reply:
I suggest maybe separating the last commit of this PR into a separate release, so if there is a big number of users of `ListUtil` or `Fields` classes, they will have more time to migrate away from them.
But if you think it's ok to delete them now, then let's just do it.

Reply:
Both of these classes reside in `org.mockito.internal`, so we don't consider them for SemVer. Therefore, we can safely delete them in any new version of Mockito.